main.c

```diff
@@ -1,78 +1,49 @@
+/* ************************** Includes ************************ */
+
 #include <stdio.h>
 #include <assert.h>
+#include "Cable_ColorCoding_Cfg.h"
 
-enum MajorColor {WHITE, RED, BLACK, YELLOW, VIOLET};
-enum MinorColor {BLUE, ORANGE, GREEN, BROWN, SLATE};
+/* ************************** Macro definitions ************************ */
 
-const char* MajorColorNames[] = {
-    "White", "Red", "Black", "Yellow", "Violet"
-};
-int numberOfMajorColors =
-    sizeof(MajorColorNames) / sizeof(MajorColorNames[0]);
-const char* MinorColorNames[] = {
-    "Blue", "Orange", "Green", "Brown", "Slate"
-};
-const int MAX_COLORPAIR_NAME_CHARS = 16;
-int numberOfMinorColors =
-    sizeof(MinorColorNames) / sizeof(MinorColorNames[0]);
+#define MAX_COLORPAIR_NAME_CHARS 12
 
-typedef struct {
-    enum MajorColor majorColor;
-    enum MinorColor minorColor;
-} ColorPair;
+/* ************************** External Functions ************************ */
+extern void ColorPairToString(const ColorPair* colorPair, char* buffer);
+extern int GetPairNumberFromColor(const ColorPair* colorPair);
+extern ColorPair GetColorFromPairNumber(int pairNumber);
+extern void PrintColorCodingManual();
 
-void ColorPairToString(const ColorPair* colorPair, char* buffer) {
-    sprintf(buffer, "%s %s",
-        MajorColorNames[colorPair->majorColor],
-        MinorColorNames[colorPair->minorColor]);
-}
+/* ************************** Function definitions ************************ */
 
-ColorPair GetColorFromPairNumber(int pairNumber) {
-    ColorPair colorPair;
-    int zeroBasedPairNumber = pairNumber - 1;
-    colorPair.majorColor = 
-        (enum MajorColor)(zeroBasedPairNumber / numberOfMinorColors);
-    colorPair.minorColor =
-        (enum MinorColor)(zeroBasedPairNumber % numberOfMinorColors);
-    return colorPair;
-}
-
-int GetPairNumberFromColor(const ColorPair* colorPair) {
-    return colorPair->majorColor * numberOfMinorColors +
-            colorPair->minorColor + 1;
+int main() {
+    testNumberToPair(4, WHITE, BROWN); /* Function call to test the pair number matching with color coding pattern */
+    testNumberToPair(5, WHITE, SLATE); /* Function call to test the pair number matching with color coding pattern */
+
+    testPairToNumber(BLACK, ORANGE, 12); /* Function call to test the Color Coding pattern matching with pair number */
+    testPairToNumber(VIOLET, SLATE, 25); /* Function call to test the Color Coding pattern matching with pair number */
+    
+    PrintColorCodingManual();  /* Function call to print the color coding user manual */
```

Comment:
All other functionality in this file is testing. Any way to test the `PrintColorCodingManual`?